spec/haskell/src/SEL4/Object/TCB.lhs

```diff
@@ -689,12 +689,14 @@ The domain cap is invoked to set the domain of a given TCB object to a given val
 >     KernelF SyscallError SchedContextInvocation
 > decodeSchedContext_YieldTo scPtr buffer = do
 >     sc <- withoutFailure $ getSchedContext scPtr
->     when (scTCB sc == Nothing) $ throw IllegalOperation
 >     ctPtr <- withoutFailure $ getCurThread
 >     when (fromJust (scTCB sc) == ctPtr) $ throw IllegalOperation
->     ct <- withoutFailure $ getObject ctPtr
+>     when (scTCB sc == Nothing) $ throw IllegalOperation
 >     priority <- withoutFailure $ threadGet tcbPriority $ fromJust $ scTCB sc
->     when (priority > tcbMCP ct) $ throw IllegalOperation
+>     ct_mcp <- withoutFailure $ threadGet tcbMCP ctPtr
+>     when (priority > ct_mcp) $ throw IllegalOperation
+>     yt_ptr <- withoutFailure $ threadGet tcbYieldTo ctPtr
+>     when (yt_ptr /= Nothing) $ throw IllegalOperation
```

Comment:
Is this a new condition to the C as well? I don't see it at the moment, but I might have missed it in a branch or PR.

Reply:
Yes, this is on Curtis's list of things to do

Reply:
Can you please include here the C change information when it becomes available? And when you do that, can you paste in the commit message as well? If it is a PR of its own, that would be good. The way things are done on the C side at the moment is very fluid and hard to keep track of. It would be good to find a better way of organising things, but that is a separate issue, I suppose.